platform/File.roc

```diff
@@ -181,3 +184,45 @@ isSymLink = \path ->
 type : Str -> Task [IsFile, IsDir, IsSymLink] [PathErr MetadataErr]
 type = \path ->
     Path.type (Path.fromStr path)
+
+
+FileReader := {readerID: U64, path: Path}
+
+## Try to create a `FileReader` for buffered (= part by part) reading given a path string.
+## See [examples/file-read-buffered.roc](https://github.com/roc-lang/basic-cli/blob/main/examples/file-read-buffered.roc) for example usage.
+##
+## This uses [rust's std::io::BufReader](https://doc.rust-lang.org/std/io/struct.BufReader.html).
+##
+## Use [readUtf8] if you want to get the entire file contents at once.
+getFileReader : Str -> Task FileReader [FileReadErr Path ReadErr]
+getFileReader = \pathStr ->
```

Comment:
There doesn't seem to be a way to close a file, is that a goal for this PR?

Reply:
That's a really good point! 😓

Reply:
Done, added a `File.closeFileReader` effect. It closes the file, and then any future "reads" will simply return NIL bytes.